doc: add minimum supported Go version to README

v0.22.0 seems to work but not v0.22.1

### What version of gazelle are you using?

v0.22.1

### What version of rules_go are you using?

v0.23.4

### What version of Bazel are you using?

3.5.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

MacOS 10.15.6 (19G2021)

### What did you do?

Followed instructions in the README.

### What did you expect to see?

gazelle succeeds

### What did you see instead?

$ bazel run //:gazelle
INFO: Repository bazel_gazelle_go_repository_tools instantiated at:
  no stack (--record_rule_instantiation_callstack not enabled)
Repository rule go_repository_tools defined at:
  /private/var/tmp/_bazel_gar/29372927b20ca936dc9ab48a0d65e05d/external/bazel_gazelle/internal/go_repository_tools.bzl:117:38: in <toplevel>
ERROR: An error occurred during the fetch of repository 'bazel_gazelle_go_repository_tools':
   Traceback (most recent call last):
	File "/private/var/tmp/_bazel_gar/29372927b20ca936dc9ab48a0d65e05d/external/bazel_gazelle/internal/go_repository_tools.bzl", line 103, column 13, in _go_repository_tools_impl
		fail("failed to build tools: " + result.stderr)
Error in fail: failed to build tools: # github.com/bazelbuild/bazel-gazelle/language/go
src/github.com/bazelbuild/bazel-gazelle/language/go/modules.go:200:6: undefined: errors.As
ERROR: /private/var/tmp/_bazel_gar/29372927b20ca936dc9ab48a0d65e05d/external/bazel_gazelle/cmd/gazelle/BUILD.bazel:10:11: no such package '@com_github_pmezard_go_difflib//difflib': no such package '@bazel_gazelle_go_repository_config//': no such package '@bazel_gazelle_go_repository_tools//': failed to build tools: # github.com/bazelbuild/bazel-gazelle/language/go
src/github.com/bazelbuild/bazel-gazelle/language/go/modules.go:200:6: undefined: errors.As
 and referenced by '@bazel_gazelle//cmd/gazelle:gazelle_lib'
ERROR: Analysis of target '//:gazelle' failed; build aborted: Analysis failed

--- a/README.rst
+++ b/README.rst
@@ -17,6 +17,7 @@ Gazelle build file generator
 .. _import_prefix: https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_library.import_prefix
 .. _strip_import_prefix: https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_library.strip_import_prefix
 .. _buildozer: https://github.com/bazelbuild/buildtools/tree/master/buildozer
+.. _Go Release Policy: https://golang.org/doc/devel/release.html#policy
 
 .. role:: cmd(code)
 .. role:: flag(code)
@@ -152,6 +153,16 @@ your root build files to avoid having to type ``-go_prefix`` every time.
 .. code:: bzl
 
   # gazelle:prefix github.com/example/project
+
+Compatibility with Go
+---------------------
+
+Gazelle is compatible with supported releases of Go, per the
+`Go Release Policy`_. The Go Team officially supports the current and previous
+minor releases. Older releases are not supported and don't receive bug fixes
+or security updates.
+
+Gazelle may use language and library features from the oldest supported release.
 
 Compatibility with rules_go
 ---------------------------

